### DISCUSSION TITLE
instance_availability_zone parameter meaning for microservices

### DISCUSSION THREAD

**Discussion by andreivaloshyn** [grafana/mimir#3045] [Help and support] [answered]
Hello!
I decided to enable `-ingester.ring.zone-awareness-enabled`
According the documentation https://grafana.com/docs/mimir/latest/operators-guide/configure/configuring-zone-aware-replication/#configuring-ingester-time-series-replication `-ingester.ring.instance-availability-zone` is required  for each pod of store-gateway explicitly which makes me to use separate k8s configmaps for each pod or some another possible workarounds.
So if I have 3 pods in StatefulSet each of them should have
`-ingester.ring.instance-availability-zone=eu-west-1b`
`-ingester.ring.instance-availability-zone=eu-west-1a`
`-ingester.ring.instance-availability-zone=eu-west-1c` depends on which node pod is running.

Or `topologySpreadConstraints` helm value can be updated to spread pods between AWS regions?

Just looking for better option.
Thank you!

**Comment by dimitarvdimitrov** [comment, marked as answer]
hi, the helm chart doesn't support multi-zone deployments yet. To support multi-zone you would need 3 separate StatefulSets each running in a separate zone via affinity rules. There is an open issue that's being actively worked on which will bring multi-zone to the chart in the coming weeks #2020

**Reply by pracucci** [reply]
If you're running Mimir with Helm, we strongly recommend to wait until multi-zone support in Helm is done (it's very close). However, let me also answer your question:

> -ingester.ring.instance-availability-zone is required for each pod of store-gateway explicitly which makes me to use separate k8s configmaps for each pod or some another possible workarounds.

CLI flags take precedence over the YAML config file. So our suggestion is to keep using the YAML config (1 config file shared across all Mimir components), but setting the availability zone as a CLI flag (so that you can easily set a different one for each per-zone StatefulSet).
